### PR TITLE
Add missing sections, connect and refactor nav bar

### PR DESCRIPTION
**Pull request by kevin-zou**
## Description
IT'S THE FINAL STRETCH. This just adds the Stats section and Resources section to the landing page. I refactored the `ContentContainer` component to take in `id` and `margin` props. `id` is used to set anchor link targets for the nav bar and `margin` is used to achieve the spacing between sections (as specified by the designs).

I set the right link targets for the nav bar and then called it a day. Smooth scrolling is supported.

### Navbar refactor
Noticed that there were two sort of bugs with the navbar:
1. Can't click through it even if it's invisible
2. Doesn't appear when you scroll up

The first problem was fixed by making the actual navbar container (ie. the one that has the `visibility` and `opacity` properties toggled) the root one of the component. Before the parent container was still "visible" and eating clicks, so you wouldn't be able to click anything on the page if it was where the navbar usually is.

The second problem I fixed by altering a bit of the `handleScroll` function. I just looked at what was done previously and it was a little different from our implementation so I just changed it to match. Now the navbar will appear on scroll up.

image
image

### CODE REVIEW

**Comment by kevin-zou** [comment]
Something I can't figure out how to accomplish is hide the navbar when a link is clicked. This was existing behaviour so I really want to get that included in this PR. My previous attempt was creating a function in the parent component:
```
const onClose = () = > {
  setShowDropdown(false);
  setOpacity(0);
  setVisibility('hidden');
}
```
and then passing that into the `MenuList` component, passing that into the `MenuItem` component, then calling it in `MenuItem.handleClick()`.

File: components/NavBar.js [hunk 1]
```diff
@@ -209,60 +230,57 @@ const NavBar = () => {
...
+        setVisibility('visible');
+        setOpacity('1');
+      } else if (scroll > lastScroll) {
+        setVisibility('hidden');
+        setOpacity('0');
+      } else {
+        setVisibility('visible');
+        setOpacity('1');
+      }
+      lastScroll = scroll;
```
Comment (phoenixthefirebird) [inline components/NavBar.js:246]: I see that I had made scroll a global variable before and that wasn't behaving as we want it. But I'm a bit confused about the line `lastScroll = scroll;` is doing: doesn't `lastScroll` get set to 0 at first anyways when the `handleScroll` function is called?

**Review by phoenixthefirebird — APPROVED** [review]

File: components/NavBar.js [hunk 2]
```diff
@@ -193,7 +214,7 @@ const NavBar = () => {
   }
 
   useEffect(() => {
-    window.addEventListener('scroll', handleScroll);
+    window.addEventListener('scroll', handleScroll());
```
[the remark this answers is not in the document]
Comment (kevin-zou) [reply components/NavBar.js]: Doesn't seem like there's a difference, I removed it in 8020c97!

File: components/NavBar.js (hunk 1 above)
Comment (kevin-zou) [reply components/NavBar.js:246]: I'm not entirely sure either, but I don't think the `var lastScroll = 0` line is being executed every time the scroll event fires. How it works in my head is:
1. scroll event fires and checks what `handleScroll` is referring to
2. `handleScroll` initializes `lastScroll` as `0` the _first_ time, and returns an anonymous function
3.  every time the scroll event fires, it executes the anonymous function, which sets `lastScroll = scroll`

**Comment by kevin-zou** [comment]
> Something I can't figure out how to accomplish is hide the navbar when a link is clicked and it scrolls up to get to the section.

Opened #91 for this

**Pull request merged**